src/dust.pegjs

```diff
@@ -115,14 +115,17 @@ identifier "identifier"
   / k:key      { var arr = ["key", k]; arr.text = k; return arr; }
 
 number "number"
-  = n:(float / integer) { return ['literal', n]; }
+  = n:(float / integer / negative_integer) { return ['literal', n]; }
 
 float "float"
-  = l:integer "." r:integer+ { return parseFloat(l + "." + r.join('')); }
+  = l:(integer / negative_integer) "." r:integer+ { return parseFloat(l + "." + r.join('')); }
 
 integer "integer"
   = digits:[0-9]+ { return parseInt(digits.join(""), 10); }
 
+negative_integer "negative_integer"
+  = sign:'-' digits:[0-9]+ { return parseInt(sign + digits.join(''), 10)}
```

Comment:
Cool, I can get on board with `signed_integer` and `unsigned_integer`.

On Thu, Mar 5, 2015 at 3:34 PM, Steven <notifications@github.com> wrote:

> In src/dust.pegjs
> <https://github.com/linkedin/dustjs/pull/549#discussion_r25913094>:
>
> >
> >  integer "integer"
> >    = digits:[0-9]+ { return parseInt(digits.join(""), 10); }
> >
> > +negative_integer "negative_integer"
> > +  = sign:'-' digits:[0-9]+ { return parseInt(sign + digits.join(''), 10)}
>
> Or float could just be integer "." r:[0-9]+. I do kind of like the
> unsigned_integer idea as well. Negative integer doesn't seem right.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/linkedin/dustjs/pull/549/files#r25913094>.
>



-- 
Seth Kinast
http://sethkinast.com/